### ISSUE TITLE
Handling Missing Values

### ISSUE THREAD

**Issue by abdulfatir** [Time-MoE/Time-MoE#4]
- Is there a "right way" to handle missing values for this model? 
- How should I pad when time series have non-uniform lengths in a batch?

**Comment by Maple728** [comment]
1. **Handling Missing Values**

Time-MoE currently does not automatically handle missing values. Please ensure that any missing data is appropriately addressed before using Time-MoE for forecasting.

2. **Batch Padding for Time Series**

You can pad any value to the beginning of the original time series and set the corresponding values in the attention mask to 0. Here's an example:

```python
mask_len = 24

attn_mask = torch.concat([torch.zeros(1, mask_len), torch.ones(1, normed_seqs.shape[1] - mask_len)], dim=1)

output = model.generate(normed_seqs, attention_mask=attn_mask, max_new_tokens=prediction_length)

normed_predictions = output[:, -prediction_length:]
```

**Comment by abdulfatir** [comment]
Thank you for your quick reply. Very Helpful! 

For missing values, would you recommend using an attention mask or imputation to ensure there are no missing values?

**Comment by Maple728** [comment]
Based on our experience, we recommend using an attention mask rather than imputation. However, we haven't yet conducted experiments to compare the forecasting performance of Time-MoE using these two methods.

Thank you for highlighting this issue. We plan to carry out experiments to compare the forecasting performance of the attention mask and imputation for handling missing values. The results will be shared in this repository once the experiments are complete.

**Comment by abdulfatir** [comment]
Thank you for sharing your insights. If I find something interesting, I will share it here with you.

**Comment by aviv-ples** [comment]
I did not fully understand how you are recommending to deal with missing data. I have nans where there's no data, I tried using the attention mask, however, it didn't work. I changed the nans to zeros and used the attention mask, however, that yielded worse performance than without.

Do you have any suggestions?
Thanks in advance

**Comment by abdulfatir** [comment]
@aviv-ples My experience dealing with missing values was similar. Although using the `attention_mask` with a dummy value (0.0) worked for me, it yielded poor results. You could try last observed value imputation (same as `ffill` in pandas) which seemed to improve things a little bit for me.

**Comment by aviv-ples** [comment]
Thank you for the quick reply. When I have missing data/zeros it means "zero activity", so there's significance to the value 0. However, having many zeros mostly causes the forecast to zero out at some point, if there isn't a clear pattern in the actual data. 

Also I wanted to ask, is there a possible way to do something similar to "num_samples" like in other models, in order to have an uncertainty interval, and a better use of WQL. Do I do it by predicting multiple times?
